toolkit/tools/pkg/isomakerlib/isomaker.go

```diff
@@ -462,8 +529,11 @@ func (im *IsoMaker) initializePaths() {
 // buildIsoImageFilePath gets the output ISO file path from the config JSON file name
 // and the image build environment.
 func (im *IsoMaker) buildIsoImageFilePath() string {
-	imageBaseName := strings.TrimSuffix(filepath.Base(im.configFilePath), ".json")
-	isoImageFileName := fmt.Sprintf("%v-%v%v.iso", imageBaseName, im.releaseVersion, im.imageNameTag)
+	isoImageFileNameSuffix := ""
+	if im.releaseVersion != "" || im.imageNameTag != "" {
+		isoImageFileNameSuffix = fmt.Sprintf("-%v%v", im.releaseVersion, im.imageNameTag)
+	}
```

Comment:
Related to my comment in `NewIsoMakerWithConfig`: maybe let's just split this into multiple calls like:
```suggestion
	imageBaseName := strings.TrimSuffix(filepath.Base(im.configFilePath), ".json")
	imageBaseName = appendImageNameSuffix(imageBaseName, im.releaseVersion)
	imageBaseName = appendImageNameSuffix(imageBaseName, im.imageNameTag)
```

And `appendImageNameSuffix` would do the check for an empty string and a hyphen.

Reply:
I've removed the code in the constructor that added the '-'. Now, this `Sprintf` is the only place where the name is constructed. I think we should be fine not adding `appendImageNameSuffix()` now.